### ISSUE TITLE
Watch pane does not use evaluateName when executing "Copy Value"

### ISSUE THREAD

**Issue by DanTup** [Microsoft/vscode#54853]
I've had a bug raised where values copied (using `Copy Value`) from the `Watch` pane are truncated. I investigated, and it seems like there is a difference in behaviour between the `Variables` and `Watch` panes. I'm examining an array with a single long string in it:

![screen shot 2018-07-23 at 9 42 59 am](https://user-images.githubusercontent.com/1078012/43066525-d008726c-8e5c-11e8-94ab-d36f14015e56.png)

In both the `scopesRequest` and the `evaluateRequest` I return a `variablesReference` for the array. In the `variablesRequest` they both return the same data, which is the following (an array of `Variable`):

```js
[
	{
		"evaluateName": "longStrings[0]",
		"indexedVariables": null,
		"name": "[0]",
		"type": "String (_OneByteString)",
		"value": "\"This is a long string that is 300 characters! This is a long string that is 300 characters! This is a long string that is 300 ch…\"",
		"variablesReference": 0
	}
]
```

This has a truncated string for display, and an `evaluateName` that will give the full string. However it seems that when I choose` Copy Value` from the `Variables` window the expression gets evaluated, but choosing `Copy Value` from the `Watch` window just copies the truncated value into the clipboard.

![screen shot 2018-07-23 at 9 47 44 am](https://user-images.githubusercontent.com/1078012/43066778-7ae6768e-8e5d-11e8-9ed2-2d89920f265f.png)

(@isidorn does this seem like a bug, or are they intended to work differently?)

**Comment by isidorn** [comment]
You are right, this is an issue and I can push a fix for it by passing variable instead of variable.value [here](https://github.com/Microsoft/vscode/blob/28a82a6eec85c81f82bfad122f3c75d12780b88d/src/vs/workbench/parts/debug/electron-browser/watchExpressionsView.ts#L180)

However this seems to uncover another bug from node-debug. Which is that the evaluateName for variables in watch window are bogus. @weinand @roblourens can you please investigate? Or if you prefer i can push a fix for this issue which will uncover the other issue

**Comment by roblourens** [comment]
I pushed a fix for the evaluateName issue.
